ui/component/socialShare/view.jsx

```diff
@@ -16,29 +19,56 @@ type Props = {
   webShareable: boolean,
   referralCode: string,
   user: any,
+  position: number,
 };
 
 function SocialShare(props: Props) {
-  const { claim, title, referralCode, user, webShareable } = props;
+  const { claim, title, referralCode, user, webShareable, position } = props;
   const [showEmbed, setShowEmbed] = React.useState(false);
   const [showExtra, setShowExtra] = React.useState(false);
+  const [includeStartTime, setincludeStartTime]: [boolean, any] = React.useState(false);
+  const [startTime, setStartTime]: [string, any] = React.useState(secondsToHms(position));
+  const [startTimeSeconds, setStartTimeSeconds]: [number, any] = React.useState(Math.floor(position));
   const isMobile = useIsMobile();
 
-  if (!claim) {
```

Comment:
Why move this? Right now it could fail below if `claim` doesn't exist 

here:
https://github.com/lbryio/lbry-desktop/pull/4142/files#diff-87e5aa61f17fe660285218396b512969R46

Reply:
I didn't move it originally, but when I tried to commit, it gave me an error saying that you can't return conditionally before calling React.useState(). So I moved it after all the React.useState() calls, and tried to make the code work if claim did happen to be null. I am not really happy with it myself, but I couldn't figure another way around the problem. This is also why I got rid of the claim destructuring, so the code wouldn't give an error if claim happened to be null. It looks like I missed one line of code that you pointed out. Can you suggest any better ways to handle the problem?

Reply:
Ah right.

I don't think you need to `useState` for the urls.

You could just do `const encodedLbryURL = generateEncodedLBRYURL(...`
Same with the `dotCom` url

Reply:
I am using state for the URLs so that the user can change the time if they want and the URLs will update automatically. Do we want to lose that functionality?

Reply:
We can still keep that functionality. If you move the URLs out of state, they will be re-created on every render. If you `setStartTime`, all of the other values can be built from that. Even the other time now that I"m looking it closer.
```
const [startTimeSeconds, setStartTimeSeconds]: [number, any] = React.useState(...)
const startTime = secondsToHms(startTimeSeconds);
const encodedUrl = generateEncodedUrl(...., includeStartTime, startTime)
const OpenDotLbryDotComUrl = generateOpenDotLbryDotComUrl(..., includeStartTime, startTime);
```

Then you can just call `setStartTimeSeconds` and everything else with update automatically. It should simplify this code quite a bit